frontend/packages/topology/integration-tests/support/step-definitions/topology/quick-search-topology.ts

```diff
@@ -15,3 +30,182 @@ Then('Display options dropdown, Filter by resource and Find by name fields are d
 Then('switch view is disabled', () => {
   cy.get(topologyPO.switcher).should('have.attr', 'aria-disabled', 'true');
 });
+
+Given('user created a workload and is at topology list view', () => {
+  navigateTo(devNavigationMenu.Add);
+  createGitWorkload(
+    'https://github.com/sclorg/nodejs-ex.git',
+    'nodejs-app',
+    'deployment',
+    'nodejs-ex-git-app',
+    false,
+  );
+  navigateTo(devNavigationMenu.Topology);
+  cy.get(topologyPO.quickSearchPO.listView).click();
+});
+
+Given('user is at topology list view', () => {
+  navigateTo(devNavigationMenu.Topology);
+  topologyPage.verifyUserIsInListView();
+});
+
+Given('user is at topology graph view', () => {
+  navigateTo(devNavigationMenu.Topology);
+  topologyPage.verifyUserIsInGraphView();
+});
+
+When('user clicks on graph view button', () => {
+  topologyPage.verifyUserIsInListView();
+  cy.get(topologyPO.quickSearchPO.graphView).click();
+});
+
+When('user clicks on list view button', () => {
+  topologyPage.verifyUserIsInGraphView();
+  cy.get(topologyPO.quickSearchPO.listView).click();
+});
+
+Then('user can see Add to project button', () => {
+  cy.get(quickSearchAddPO.quickSearchButton).should('be.visible');
+});
+
+When('user clicks Add to project button', () => {
+  cy.get(quickSearchAddPO.quickSearchButton)
+    .should('be.visible')
+    .click();
+});
+
+Then('user can see Add to project search bar', () => {
+  cy.get(quickSearchAddPO.quickSearchBar).should('be.visible');
+});
+
+Then('user enters {string} in Add to project search bar', (node: string) => {
```

Comment:
there is a bunch of mismatched step definitions, like this one is supposed to be a `When` step
make sure the tasks match the feature file